### PR TITLE
ECSOPS-49: Bundle properties files in the jar.

### PR DESCRIPTION
**Pull request by magical**
This is enabled by putting the properties files in src/main/resources
and using Class.getResourceAsStream to read the properties, and makes
deployment a bit easier because we have fewer files to copy.

While we're here, clean up a couple things
- Resource.loadProperties takes a file name but it's always called with
  'resource.properties' and we shouldn't burden callers with needing to
  know where the resource properties are located.
- Let's read build.properties in InfoResource instead of Info,
  making Info a POJO class just like Error.

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy [hunk 1]
```diff
@@ -14,6 +14,21 @@ import javax.ws.rs.core.Response
 class InfoResource extends Resource {
     private Info info = new Info()
 
+    public InfoResource() {
+        def stream = this.class.getResourceAsStream('/build.properties')
+        if (stream == null) {
+            throw new Exception("couldn't load build.properties")
+        }
```
Comment (floppyears) [inline src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy:21]: A couple of things: 
- the check can be changed to: if (!stream) to make the code shorter
- have you tested whether the exception will prevent the startup of the program? An error should be thrown, but probably not prevent the application from running

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy [hunk 2]
```diff
@@ -9,8 +9,12 @@ import javax.ws.rs.core.Response.ResponseBuilder
...
     protected static Properties properties = new Properties()
 
-    public static loadProperties(String fileName) {
-        properties.load(new FileReader(fileName))
+    public static loadProperties() {
+        def stream = this.getResourceAsStream('resource.properties')
+        if (stream == null) {
+            throw new Exception("couldn't open resource.properties")
+        }
+        properties.load(stream)
```
Comment (floppyears) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy:17]: This logic is a bit repetitive with the InfoResource. The InfoResource inherits this class and could just reuse the logic.

File: src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy (hunk 1 above)
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy:21]: - (!stream) has different semantics
- yes, the program will not start if build.properties (or resource.properties) is not found. but that isn't anything new; the only difference is it used to throw a FileNotFoundException. do we really want to try to start up without the build info? now that it's bundled in the jar it shouldn't really ever be missing

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy (hunk 2 above)
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy:17]: except we don't want InfoResource to overwrite the properties member

Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy:17]: that said, i guess i could turn this into a more general helper method

**Pull request merged**